synapse/handlers/federation_event.py

```diff
@@ -1031,6 +1031,12 @@ async def _get_state_ids_after_missing_prev_event(
             InvalidResponseError: if the remote homeserver's response contains fields
                 of the wrong type.
         """
+
+        # It would be better if we could query the difference from our known
+        # state to the given `event_id` so the sending server doesn't have to
+        # send as much and we don't have to process as many events. For example
+        # in a room like #matrixhq, we get 200k events (77k state_events, 122k
```

Comment:
either Matrix HQ or #matrix:matrix.org, but not this half-way house :)

```suggestion
        # in a room like #matrix:matrix.org, we get 200k events (77k state_events, 122k
```